charts/tfy-karpenter-config/files/al2-soci-gpu-provisioner-userdata.sh

```diff
@@ -1,130 +1,57 @@
---retry-delay 1 \
--L https://github.com/awslabs/soci-snapshotter/releases/download/v${SOCI_RELEASE_VERSION}/soci-snapshotter-${SOCI_RELEASE_VERSION}-linux-amd64.tar.gz \
--o $SOCI_TAR_FILENAME
-echo "$SOCI_TAR_CHECKSUM $SOCI_TAR_FILENAME" | sha256sum -c
-
-tar -C /usr/local/bin -xvf $SOCI_TAR_FILENAME soci soci-snapshotter-grpc
-rm $SOCI_TAR_FILENAME
-
-cat > $SOCI_SYSTEMD_SERVICE_FILEPATH << EOF
-[Unit]
-Description=soci snapshotter containerd plugin
-Documentation=https://github.com/awslabs/soci-snapshotter
-After=network.target
-Before=containerd.service
-
-[Service]
-Type=notify
-ExecStart=/usr/local/bin/soci-snapshotter-grpc
-Restart=always
-RestartSec=5
-
-[Install]
-WantedBy=multi-user.target
-EOF
-
-mkdir -p $SOCI_CONFIG_DIR
 
-cat > $SOCI_CONFIG_FILEPATH << EOF
+# --- Install soci-snapshotter ---
+pushd /tmp
+curl -sSL -o "$ARCHIVE" "https://github.com/awslabs/soci-snapshotter/releases/download/v$VERSION/$ARCHIVE"
+curl -sSL -o "$ARCHIVE.sha256sum" "https://github.com/awslabs/soci-snapshotter/releases/download/v$VERSION/$ARCHIVE.sha256sum"
+sha256sum -c "$ARCHIVE.sha256sum"
+tar -xvzf "$ARCHIVE" -C /usr/local/bin soci-snapshotter-grpc
+rm -f "$ARCHIVE" "$ARCHIVE.sha256sum"
+popd
+
+# --- Configure soci-snapshotter ---
+mkdir -p /etc/soci-snapshotter-grpc
+cat <<EOF > /etc/soci-snapshotter-grpc/config.toml
 [cri_keychain]
-enable_keychain=true
-image_service_path="/run/containerd/containerd.sock"
+enable_keychain = true
+image_service_path = "/run/containerd/containerd.sock"
 EOF
 
+# --- Enable and start soci-snapshotter ---
+curl -sSL -o /etc/systemd/system/soci-snapshotter.service "https://raw.githubusercontent.com/awslabs/soci-snapshotter/v$VERSION/soci-snapshotter.service"
+systemctl daemon-reexec
 systemctl daemon-reload
 systemctl enable --now soci-snapshotter
-systemctl status soci-snapshotter
-
-echo patching $CONTAINERD_CONFIG_FILEPATH
-cat > $CONTAINERD_CONFIG_FILEPATH << EOF
-version = 2
-root = "/var/lib/containerd"
-state = "/run/containerd"
-imports = ["/etc/containerd/config.d/*.toml"]
-
-[grpc]
-address = "/run/containerd/containerd.sock"
 
+# --- Configure containerd to use soci snapshotter ---
+cat > /etc/containerd/config.d/soci.toml <<EOF
 [plugins."io.containerd.grpc.v1.cri".containerd]
-default_runtime_name = "runc"
+default_runtime_name = "nvidia"
 discard_unpacked_layers = true
-## ENABLE SOCI SNAPSHOTTER
 snapshotter = "soci"
 disable_snapshot_annotations = false
-##
-
-[plugins."io.containerd.grpc.v1.cri"]
-sandbox_image = "SANDBOX_IMAGE"
+[proxy_plugins]
+[proxy_plugins.soci]
+address = '/run/soci-snapshotter-grpc/soci-snapshotter-grpc.sock'
+type = 'snapshot'
 
-[plugins."io.containerd.grpc.v1.cri".registry]
-config_path = "/etc/containerd/certs.d:/etc/docker/certs.d"
+[proxy_plugins.soci.exports]
+root = '/var/lib/soci-snapshotter-grpc'
 
-[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.runc]
+[plugins."io.containerd.grpc.v1.cri".containerd.runtimes.nvidia]
 runtime_type = "io.containerd.runc.v2"
```

Comment:
Same question here - Is this necessarily needed? I am assuming this will already be present eks' containerd config.toml

Reply:
Without setting this it doesn't work for me, the gpu-operator pods are not able to find `nvidia` runtime